recipes/emsdk/all/conanfile.py

```diff
@@ -0,0 +1,140 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+import json
+
+required_conan_version = ">=1.33.0"
+
+
+class EmSDKConan(ConanFile):
+    name = "emsdk"
+    description = "Emscripten SDK. Emscripten is an Open Source LLVM to JavaScript compiler"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/kripken/emscripten"
+    topics = ("emsdk", "emscripten", "sdk")
+    license = "MIT"
+    settings = "os", "arch"
+
+    short_paths = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def requirements(self):
+        self.requires("nodejs/16.3.0")
+        # self.requires("python")  # FIXME: Not available as Conan package
+        # self.requires("wasm")  # FIXME: Not available as Conan package
+
+    def validate(self):
+        if hasattr(self, "settings_target") and self.settings_target:
+            if self.settings_target.os != "Emscripten":
+                raise ConanInvalidConfiguration("When using {}, target os must be Emscripten".format(self.name))
+
+    def package_id(self):
+        del self.info.settings.os
```

Comment:
```suggestion
        pass
        #del self.info.settings.os
```

Reply:
it causes to run incorrect executable - ELF is consumed on Mac, it shouldn't happen